lib/honeykiq/server_middleware.rb

```diff
@@ -47,9 +55,12 @@ def queue_fields(queue)
       }
     end
 
-    def duration_ms(event)
+    def run_event(event, msg, queue_name)
+      event.add(**default_fields(msg, queue_name))
+      event.add(**before_fields)
       start_time = Time.now
       yield
+      event.add(**after_fields)
```

Comment:
I am wondering whether we might want to add `after_fields` within the `ensure` in case a call fails?

Reply:
that is a good shout, thanks!